src/tests/mod.rs

```diff
@@ -1088,3 +1089,63 @@ fn opt_out_from_default_substitutes() {
 
     assert_eq!(code.to_string(), expected_code.to_string());
 }
+
+#[test]
+fn ensure_unique_type_paths_test() {
+    use scale_info::PortableRegistry;
+
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    #[scale_info(skip_type_params(Hash))]
+    struct Header<Hash, AccountId> {
+        id: AccountId,
+        hash: Hash,
+    }
+
+    let mut registry = Testgen::new()
+        .with::<Header<String, u32>>()
+        .with::<Header<u8, u32>>()
+        .with::<Header<u16, u32>>()
+        .into_portable_registry();
+
+    let sorted_type_paths = |registry: &PortableRegistry| -> Vec<String> {
+        let mut lines = registry
+            .types
+            .iter()
+            .map(|t| t.ty.path.segments.clone().join("::"))
+            .collect::<Vec<String>>();
+        lines.sort();
+        lines
+    };
+
+    let e1 = sorted_type_paths(&registry);
+    assert_eq!(
+        e1,
+        vec![
+            "",
```

Comment:
I don't follow where `""`comes from? 
Is it possible to remove or intended (I don't follow)?

Reply:
Oh these lines are for the 4 primitive types registered in the TypeRegistry. It is probably better if I filter them out in this test.